### PR TITLE
Tooltip animation

### LINKED ISSUE #3213: [Enhance] Add Animation to Tooltip - All Frameworks
**User Story:**
As Spark we want Animation to work with our Angular tooltip component.

Example: https://webdesign.tutsplus.com/tutorials/css-tooltip-magic--cms-28082

**Notes:**
This is a follow up from: https://github.com/sparkdesignsystem/spark-design-system/pull/3170
SME: Bob

**AC:**
- Angular Tooltip will use animation

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
Adds animation to the Tooltip component.

### Associated Issue
Fixes #3213 

### Code
 - [X] Build Component in HTML
 - [X] Build Component in Angular
 - [X] Build Component in React
 - [X] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)
 - [X] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [X] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [X] Google Chrome
  - [x] Google Chrome (Mobile)
  - [X] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [X] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [X] Apple Safari
  - [x] Apple Safari (Mobile)
  
### Accessibility Testing
  - [x] Axe browser extension
  - [X] Jaws Inspect
  - [x] VoiceOver (iOS) or Talkback (Android)

### Deploy Preview Reviewed and Approved
 - [X] Design Team
 - [x] Accessibility Team

### Screenshots
Add screenshots to help explain your PR if you'd like. However, this is not
expected.

### CODE REVIEW

File: html/components/tooltip.stories.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default {
   decorators: [(story) => `<div class="sprk-o-Box">${story()}</div>`],
   parameters: {
     info: `
-${markdownDocumentationLinkBuilder('tooltips')}
+${markdownDocumentationLinkBuilder('toggle')}
```
Comment (guern1kn) [inline html/components/tooltip.stories.js:11]: Should this have gotten changed?

File: html/components/_tooltip.scss [hunk 2]
```diff
@@ -62,16 +59,16 @@
     -webkit-transform: rotateY(0deg) rotate(45deg);
     -ms-transform: rotateY(0deg) rotate(45deg);
     transform: rotateY(0deg) rotate(45deg);
-    content: " ";
+    content: ' ';
```
Comment (RVMendoza) [inline html/components/_tooltip.scss:62]: do we use single quotes in CSS?

Comment (bobwise) [reply html/components/_tooltip.scss:62]: I see a mix of single and double in the code right now. This change was made by one of the linters.

File: html/components/_tooltip.scss [hunk 3]
```diff
@@ -132,5 +129,34 @@
...
 .sprk-c-Tooltip__trigger.sprk-c-Tooltip--toggled + .sprk-c-Tooltip,
 .sprk-c-Tooltip:hover {
-  display: inline-block;
+  opacity: 1;
+  visibility: visible;
+
+  &.sprk-c-Tooltip--top-left {
+    top: auto;
+    right: -1rem;
+    bottom: 2rem;
```
[the remark this answers is not in the document]
Comment (bobwise) [reply html/components/_tooltip.scss:140]: Maybe at some point in the future, but I don't think it makes sense right now. Currently Spark Tooltip always appears on a Spark Icon of a specific size. Changing any of these properties would result in a Tooltip that isn't pointing at its trigger, which is not very usable.

image

**Pull request merged**